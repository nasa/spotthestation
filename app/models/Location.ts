import { types } from "mobx-state-tree"
import { Sighting } from "./Sightings"

export const Point = types.model("Point", {
  lat: types.optional(types.number, 0),
  lng: types.optional(types.number, 0),
})

export const Location = types.model("Location", {
  title: types.optional(types.string, ""),
  subtitle: types.optional(types.string, ""),
  location: types.optional(Point, {}),
  sightings: types.optional(types.array(Sighting), []),
  lastSightingOrbitPointAt: types.maybeNull(types.string),
  lastUpdatedAt: types.maybeNull(types.string),
  filterTimeOfDay: types.optional(types.string, ""),
  filterDuration: types.optional(types.string, ""),
  googlePlaceId: types.optional(types.string, ""),
  osmPlaceId: types.optional(types.string, ""),
  timezone: types.optional(types.string, ""),
})
