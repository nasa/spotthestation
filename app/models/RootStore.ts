import { Instance, SnapshotOut, types } from "mobx-state-tree"
import RootStoreActions from "./actions"
import { Location } from "./Location"
import { Modal } from "./Modal"

/**
 * A RootStore model.
 */
export const RootStoreModel = types
  .model("RootStore", {
    savedLocations: types.optional(types.array(Location), []),
    selectedLocation: types.maybeNull(Location),
    currentLocation: types.maybeNull(Location),
    issData: types.optional(types.frozen(), []),
    initLoading: types.optional(types.boolean, false),
    trajectoryError: types.optional(types.boolean, false),
    trajectoryErrorKind: types.optional(types.string, ""),
    issDataLoaded: types.optional(types.boolean, false),
    sightingsLoaded: types.optional(types.boolean, false),
    currentModal: types.maybeNull(Modal),
    modalsQueue: types.optional(types.array(types.string), []),
    isCurrentLocationUpdating: types.optional(types.boolean, false),
  })
  .props({})
  .actions(RootStoreActions)

/**
 * The RootStore instance.
 */
export interface RootStore extends Instance<typeof RootStoreModel> {}
/**
 * The data of a RootStore.
 */
export interface RootStoreSnapshot extends SnapshotOut<typeof RootStoreModel> {}
