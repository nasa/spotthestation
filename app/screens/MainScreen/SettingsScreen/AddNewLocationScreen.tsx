import { useNavigation, useRoute } from "@react-navigation/native"
import { observer } from "mobx-react-lite"
import React, { useCallback, useRef, useState } from "react"
import { ViewStyle, TextStyle, View, TextInput } from "react-native"
import { useSafeAreaInsets } from "react-native-safe-area-context"
import { Accessory, Button, Icon, Screen, Text, TextField } from "../../../components"
import { colors, spacing, typography } from "../../../theme"
import { IconLinkButton } from "../../OnboardingScreen/components/IconLinkButton"
import { translate } from "../../../i18n/translate"
import Snackbar from "react-native-snackbar"
import { useStores } from "../../../models"
import { StyleFn, useStyles } from "../../../utils/useStyles"
import { LocationType } from "../../../services/api"
import { LocationAutocomplete } from "../../../components/LocationAutocomplete"

export interface AddNewLocationScreenParams {
  defaultLocation?: LocationType
}

export const AddNewLocationScreen = observer(function AddNewLocationScreen() {
  const {
    $headerStyleOverride,
    $container,
    $topButtonsContainer,
    $buttonsContainer,
    $scrollContainer,
    $button,
    $title,
    $locations,
    $active,
    $locationsListContainer,
    $dropdownSelected,
    $locationsRow,
    $locationsRowText,
    $dropdownText,
    $save,
    $saveText,
    $saveDisabled,
  } = useStyles(styles)

  const navigation = useNavigation()
  const { savedLocations, setSavedLocations, setNewSavedLocation } = useStores()
  const topInset = useSafeAreaInsets().top
  const { defaultLocation } = useRoute().params as AddNewLocationScreenParams

  const addressRef = useRef<TextInput>()
  const [isFocus, setIsFocus] = useState(false)
  const [textValue, setTextValue] = useState("")
  const [titleValue, setTitleValue] = useState(defaultLocation?.title || "")
  const [location, setLocation] = useState<LocationType>(
    defaultLocation ? null : { ...defaultLocation },
  )

  const handleClear = () => {
    addressRef.current?.clear()
    setTextValue("")
    setLocation(null)
  }

  const handleNavigate = () =>
    navigation.navigate("LocationSettings" as never, { update: Date.now() } as never)

  const handleSave = useCallback(() => {
    location.title = titleValue || (location?.subtitle ?? "").split(",")[0]
    if (!location.title || !location.location) return
    let res: LocationType[] = [...savedLocations]
    if (res.find((item) => item.title === titleValue)) {
      Snackbar.show({
        text: translate("snackBar.locationExist"),
        duration: Snackbar.LENGTH_LONG,
        action: {
          text: translate("snackBar.ok"),
          textColor: "green",
          onPress: () => {
            Snackbar.dismiss()
          },
        },
      })
      return
    }

    if (defaultLocation) {
      res = res.filter((item) => item.title !== defaultLocation.title)
      res.push(location)
      setSavedLocations(res)
    } else {
      setNewSavedLocation(location).catch((error) => {
        Snackbar.show({
          text: error,
          duration: Snackbar.LENGTH_LONG,
          action: {
            text: translate("snackBar.ok"),
            textColor: "green",
            onPress: () => {
              Snackbar.dismiss()
            },
          },
        })
      })
    }

    Snackbar.show({
      text: translate("snackBar.locationSaved"),
      duration: Snackbar.LENGTH_LONG,
      action: {
        text: translate("snackBar.ok"),
        textColor: "green",
        onPress: () => {
          Snackbar.dismiss()
        },
      },
    })
    handleNavigate()
  }, [titleValue, location, savedLocations])

  const headerStyle = { ...$headerStyleOverride }
  headerStyle.top = Number(headerStyle.top) + topInset

  const isSaveDisabled = !location?.location || !location?.title

  return (
    <Screen
      preset="fixed"
      contentContainerStyle={[$container, headerStyle]}
      style={{ backgroundColor: colors.palette.neutral900 }}
      statusBarStyle="light"
    >
      <View
        accessible
        accessibilityLabel="Location settings scrollable us area"
        accessibilityHint="Location settings scrollable us area"
        accessibilityRole="scrollbar"
        style={$scrollContainer}
      >
        <View style={$topButtonsContainer}>
          <IconLinkButton
            icon="x"
            buttonStyle={$button}
            iconColor={colors.palette.neutral250}
            iconSize={20}
            onPress={() => navigation.goBack()}
          />
          {!defaultLocation && (
            <IconLinkButton
              icon="map"
              buttonStyle={$button}
              iconColor={colors.palette.neutral250}
              iconSize={20}
              onPress={() => navigation.navigate("AddNewLocationMap" as never)}
            />
          )}
        </View>
        <Text
          tx={`settings.locationSettingsData.addNewLocation.${
            defaultLocation ? "generalTitleEdit" : "generalTitleAdd"
          }`}
          style={$title}
        />
        <LocationAutocomplete
          ref={addressRef}
          placeholder={translate(
            "settings.locationSettingsData.addNewLocation.searchInputPlaceholder",
          )}
          onPress={(data) => {
            setLocation({
              title: data.name,
              subtitle: data.display_name,
              location: { lat: Number(data.lat), lng: Number(data.lon) },
              sightings: [],
            })
            setTitleValue(data.name)
          }}
          styles={{
            textInputContainer: isFocus ? [$locations, $active] : $locations,
            textInput: {
              ...$dropdownText,
              ...$dropdownSelected,
            },
            listView: $locationsListContainer,
            row: $locationsRow,
            separator: { height: 0 },
            container: { flex: 0 },
          }}
          textInputProps={{
            allowFontScaling: false,
            placeholderTextColor: colors.palette.neutral450,
            value: location?.subtitle || textValue,
            onFocus: () => setIsFocus(true),
            onBlur: () => setIsFocus(false),
            onChangeText: (text) => setTextValue(text),
            clearButtonMode: "never",
          }}
          renderRow={({ description }) => {
            return (
              <Text
                text={description}
                style={$locationsRowText}
                ellipsizeMode="tail"
                numberOfLines={1}
              />
            )
          }}
          renderLeftAccessory={({ style }) => (
            <Icon icon="pin" size={28} color={colors.palette.neutral450} style={style} />
          )}
          renderRightAccessory={({ style }) =>
            isFocus &&
            textValue && (
              <Icon
                icon="xCircle"
                size={28}
                color={colors.palette.neutral450}
                style={style}
                onPress={handleClear}
              />
            )
          }
        />
        <TextField
          accessible
          accessibilityLabel="location title input"
          accessibilityHint="location title input"
          accessibilityRole="text"
          value={titleValue}
          autoCapitalize="none"
          autoCorrect={false}
          placeholderTx="settings.locationSettingsData.addNewLocation.nameInputPlaceholder"
          onChangeText={setTitleValue}
          inputWrapperStyle={$locations}
          renderLeftAccessory={({ style }) => (
            <Accessory icon="save" color={colors.palette.neutral450} style={style} />
          )}
        />
        <View style={$buttonsContainer}>
          <Button
            testID="saveButton"
            accessible
            accessibilityLabel="save button"
            accessibilityHint="save location"
            tx="settings.locationSettingsData.addNewLocation.saveButton"
            disabled={isSaveDisabled}
            style={[$save, isSaveDisabled && $saveDisabled]}
            textStyle={$saveText}
            pressedStyle={$save}
            onPress={handleSave}
          />
        </View>
      </View>
    </Screen>
  )
})

const styles: StyleFn = ({ scale, fontSizes, lineHeights }) => {
  const $headerStyleOverride: TextStyle = {
    top: scale(24),
  }

  const $container: ViewStyle = {
    flex: 1,
    backgroundColor: colors.backgroundDark,
    height: "100%",
  }

  const $topButtonsContainer: ViewStyle = {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    marginBottom: scale(36),
  }

  const $buttonsContainer: ViewStyle = {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
  }

  const $scrollContainer: ViewStyle = {
    paddingHorizontal: scale(18),
  }

  const $button: ViewStyle = {
    backgroundColor: colors.palette.neutral550,
    width: scale(42),
    height: scale(42),
  }

  const $text: TextStyle = {
    fontFamily: typography.primary?.normal,
    fontSize: fontSizes[18],
    lineHeight: lineHeights[22],
    color: colors.palette.neutral450,
    textAlign: "left",
    paddingBottom: scale(24),
  }

  const $title: TextStyle = {
    ...$text,
    fontSize: fontSizes[36],
    lineHeight: lineHeights[44],
    color: colors.palette.neutral250,
  }

  const $locations: ViewStyle = {
    borderWidth: scale(1.5),
    borderColor: "transparent",
    borderRadius: scale(28),
    height: scale(56),
    backgroundColor: colors.palette.neutral350,
    overflow: "hidden",
    marginBottom: scale(18),
  }

  const $active: ViewStyle = {
    borderWidth: scale(1.5),
    borderColor: colors.palette.buttonBlue,
    backgroundColor: colors.palette.overlayBlue,
  }

  const $locationsListContainer: ViewStyle = {
    borderRadius: scale(12),
    backgroundColor: colors.palette.neutral550,
    overflow: "hidden",
    width: "85%",
    alignSelf: "center",
    marginTop: scale(3),
  }

  const $dropdownLeftAccessory: ViewStyle = {
    marginStart: scale(spacing.large),
    height: scale(56),
    justifyContent: "center",
    alignItems: "center",
  }

  const $dropdownRightAccessory: ViewStyle = {
    marginEnd: scale(spacing.large),
    height: scale(56),
    justifyContent: "center",
    alignItems: "center",
  }

  const $dropdownSelected: TextStyle = {
    color: colors.palette.neutral250,
  }

  const $locationsRow: TextStyle = {
    backgroundColor: "transparent",
    paddingHorizontal: scale(spacing.large),
  }

  const $locationsRowText: TextStyle = {
    fontSize: fontSizes[18],
    lineHeight: lineHeights[22],
    color: colors.palette.neutral250,
  }

  const $dropdownText: TextStyle = {
    flex: 1,
    height: scale(56),
    alignSelf: "stretch",
    fontFamily: typography.primary.normal,
    fontSize: fontSizes[18],
    lineHeight: lineHeights[22],
    paddingHorizontal: 0,
    marginHorizontal: scale(spacing.small),
    textAlignVertical: "center",
    borderRadius: 0,
    backgroundColor: "transparent",
  }

  const $save: ViewStyle = {
    width: "100%",
    height: scale(56),
    backgroundColor: colors.palette.buttonBlue,
    borderRadius: scale(28),
    borderWidth: 0,
    marginVertical: scale(24),
  }

  const $saveDisabled: ViewStyle = {
    opacity: 0.5,
  }

  const $saveText: TextStyle = {
    color: colors.palette.neutral100,
    fontSize: fontSizes[18],
    fontFamily: typography.primary.medium,
    lineHeight: lineHeights[21],
  }

  return {
    $headerStyleOverride,
    $container,
    $topButtonsContainer,
    $buttonsContainer,
    $scrollContainer,
    $button,
    $text,
    $title,
    $locations,
    $active,
    $locationsListContainer,
    $dropdownLeftAccessory,
    $dropdownRightAccessory,
    $dropdownSelected,
    $locationsRow,
    $locationsRowText,
    $dropdownText,
    $save,
    $saveText,
    $saveDisabled,
  }
}
