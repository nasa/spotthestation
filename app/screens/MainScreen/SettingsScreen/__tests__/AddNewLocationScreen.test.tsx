import React from "react"
import { AddNewLocationScreen } from "../AddNewLocationScreen"
import { NavigationContainer } from "@react-navigation/native"
import { act, fireEvent, render, waitFor } from "@testing-library/react-native"
import { useStores } from "../../../../models"

jest.mock("@react-navigation/native", () => ({
  ...Object.assign({}, jest.requireActual("@react-navigation/native")),
  useRoute: () => ({
    params: {
      id: 1,
      defaultLocation: null,
    },
  }),
}))
it("renders correctly", async () => {
  const component = render(
    <NavigationContainer>
      <AddNewLocationScreen />
    </NavigationContainer>,
  )

  await act(async () => {
    await new Promise((resolve) => setTimeout(resolve, 100))
  })

  expect(component.toJSON()).toMatchSnapshot()
})

test("clears the text input and sets location value after selecting a place from the list", async () => {
  const { getByPlaceholderText } = render(
    <NavigationContainer>
      <AddNewLocationScreen />
    </NavigationContainer>,
  )
  const textInput = getByPlaceholderText("homeScreen.selectLocation.inputPlaceholder undefined")

  await waitFor(() => {
    expect(textInput.props.value).toBe("")
  })
})

jest.mock("../../../../models", () => ({
  useStores: jest.fn(() => ({
    savedLocations: [],
    setSavedLocations: jest.fn(),
    setNewSavedLocation: jest.fn(),
    handleSave: jest.fn(),
  })),
}))

test("calls handleSave when Save button is clicked", async () => {
  const { getByPlaceholderText } = render(
    <NavigationContainer>
      <AddNewLocationScreen />
    </NavigationContainer>,
  )
  const { handleSave }: any = useStores()

  const textInput = getByPlaceholderText("homeScreen.selectLocation.inputPlaceholder undefined")
  fireEvent.changeText(textInput, "Some location")

  await waitFor(() => {
    expect(handleSave).toHaveBeenCalledTimes(0)
  })
})
