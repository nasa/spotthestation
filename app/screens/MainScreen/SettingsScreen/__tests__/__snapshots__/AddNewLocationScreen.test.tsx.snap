// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <View
    accessibilityHint="Location settings scrollable us area"
    accessibilityLabel="Location settings scrollable us area"
    accessibilityRole="scrollbar"
    accessible={true}
    style={
      {
        "paddingHorizontal": 26,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 52,
        }
      }
    >
      <View
        accessibilityRole="imagebutton"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          animatedStyle={
            {
              "value": {},
            }
          }
          collapsable={false}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#353739",
              "borderRadius": 143,
              "display": "flex",
              "height": 60,
              "justifyContent": "center",
              "overflow": "hidden",
              "position": "relative",
              "width": 60,
            }
          }
        >
          <View
            style={
              [
                {
                  "backgroundColor": "transparent",
                },
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#1E2022",
                    "borderRadius": 143,
                    "display": "flex",
                    "height": 60,
                    "justifyContent": "center",
                    "overflow": "hidden",
                    "position": "relative",
                    "width": 60,
                  },
                  undefined,
                ],
              ]
            }
          >
            <ViewManagerAdapter_ExpoBlurView
              proxiedProperties={
                {
                  "intensity": 0,
                  "tint": "default",
                }
              }
              style={
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            />
            <View>
              <RCTImageView
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                internal_analyticTag={null}
                resizeMode="contain"
                source={
                  [
                    {
                      "height": undefined,
                      "uri": undefined,
                      "width": undefined,
                    },
                  ]
                }
                style={
                  {
                    "height": 29,
                    "overflow": "hidden",
                    "resizeMode": "contain",
                    "tintColor": "#DEDDDE",
                    "width": 29,
                  }
                }
                tintColor="#DEDDDE"
              />
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityRole="imagebutton"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          animatedStyle={
            {
              "value": {},
            }
          }
          collapsable={false}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#353739",
              "borderRadius": 143,
              "display": "flex",
              "height": 60,
              "justifyContent": "center",
              "overflow": "hidden",
              "position": "relative",
              "width": 60,
            }
          }
        >
          <View
            style={
              [
                {
                  "backgroundColor": "transparent",
                },
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#1E2022",
                    "borderRadius": 143,
                    "display": "flex",
                    "height": 60,
                    "justifyContent": "center",
                    "overflow": "hidden",
                    "position": "relative",
                    "width": 60,
                  },
                  undefined,
                ],
              ]
            }
          >
            <ViewManagerAdapter_ExpoBlurView
              proxiedProperties={
                {
                  "intensity": 0,
                  "tint": "default",
                }
              }
              style={
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            />
            <View>
              <RCTImageView
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                internal_analyticTag={null}
                resizeMode="contain"
                source={
                  [
                    {
                      "height": undefined,
                      "uri": undefined,
                      "width": undefined,
                    },
                  ]
                }
                style={
                  {
                    "height": 29,
                    "overflow": "hidden",
                    "resizeMode": "contain",
                    "tintColor": "#DEDDDE",
                    "width": 29,
                  }
                }
                tintColor="#DEDDDE"
              />
            </View>
          </View>
        </View>
      </View>
    </View>
    <Text
      allowFontScaling={false}
      style={
        [
          {},
          [
            {
              "fontSize": 23,
              "lineHeight": 35,
            },
            {
              "fontFamily": "Helvetica Neue",
            },
            {
              "color": "#191015",
            },
          ],
          undefined,
          undefined,
          {
            "color": "#DEDDDE",
            "fontFamily": "Helvetica Neue",
            "fontSize": 52,
            "lineHeight": 63,
            "paddingBottom": 35,
            "textAlign": "left",
          },
        ]
      }
    >
      settings.locationSettingsData.addNewLocation.generalTitleAdd undefined
    </Text>
    <View
      collapsable={false}
      style={
        {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "flex-start",
                "backgroundColor": "#1E2022",
                "borderRadius": 40,
                "flexDirection": "row",
                "height": 80,
                "overflow": "hidden",
              },
              false,
              undefined,
              {
                "paddingStart": 0,
              },
              {
                "paddingEnd": 0,
              },
              {
                "backgroundColor": "#1E2022",
                "borderColor": "transparent",
                "borderRadius": 40,
                "borderWidth": 2,
                "height": 80,
                "marginBottom": 26,
                "overflow": "hidden",
              },
            ]
          }
        >
          <View>
            <RCTImageView
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              internal_analyticTag={null}
              resizeMode="contain"
              source={
                [
                  {
                    "height": undefined,
                    "uri": undefined,
                    "width": undefined,
                  },
                ]
              }
              style={
                {
                  "alignItems": "center",
                  "height": 80,
                  "justifyContent": "center",
                  "marginStart": 35,
                  "overflow": "hidden",
                  "resizeMode": "contain",
                  "tintColor": "#ADADAE",
                  "width": 40,
                }
              }
              tintColor="#ADADAE"
            />
          </View>
          <TextInput
            accessibilityHint="type for search location"
            accessibilityLabel="search location"
            accessibilityRole="search"
            accessible={true}
            allowFontScaling={false}
            autoCapitalize="none"
            autoCorrect={false}
            clearButtonMode="never"
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="homeScreen.selectLocation.inputPlaceholder undefined"
            placeholderTextColor="#ADADAE"
            style={
              [
                {
                  "alignSelf": "stretch",
                  "color": "#DEDDDE",
                  "flex": 1,
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 26,
                  "height": 80,
                  "marginHorizontal": 17,
                  "paddingHorizontal": 0,
                  "paddingVertical": 0,
                  "textAlignVertical": "center",
                },
                false,
                undefined,
                undefined,
                {
                  "alignSelf": "stretch",
                  "backgroundColor": "transparent",
                  "borderRadius": 0,
                  "color": "#DEDDDE",
                  "flex": 1,
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 26,
                  "height": 80,
                  "lineHeight": 32,
                  "marginHorizontal": 17,
                  "paddingHorizontal": 0,
                  "textAlignVertical": "center",
                },
              ]
            }
            textAlignVertical="top"
            underlineColorAndroid="rgba(0, 0, 0, 0)"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "alignItems": "flex-start",
              "backgroundColor": "#1E2022",
              "borderRadius": 40,
              "flexDirection": "row",
              "height": 80,
              "overflow": "hidden",
            },
            false,
            undefined,
            {
              "paddingStart": 0,
            },
            undefined,
            {
              "backgroundColor": "#1E2022",
              "borderColor": "transparent",
              "borderRadius": 40,
              "borderWidth": 2,
              "height": 80,
              "marginBottom": 26,
              "overflow": "hidden",
            },
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "height": 80,
              "justifyContent": "center",
              "marginStart": 35,
            }
          }
        >
          <RCTImageView
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            internal_analyticTag={null}
            resizeMode="contain"
            source={
              [
                {
                  "height": undefined,
                  "uri": undefined,
                  "width": undefined,
                },
              ]
            }
            style={
              {
                "height": 40,
                "overflow": "hidden",
                "resizeMode": "contain",
                "tintColor": "#ADADAE",
                "width": 40,
              }
            }
            tintColor="#ADADAE"
          />
        </View>
        <TextInput
          accessibilityHint="location title input"
          accessibilityLabel="location title input"
          accessibilityRole="text"
          accessible={true}
          allowFontScaling={false}
          autoCapitalize="none"
          autoCorrect={false}
          editable={true}
          onChangeText={[Function]}
          placeholder="settings.locationSettingsData.addNewLocation.nameInputPlaceholder undefined"
          placeholderTextColor="#ADADAE"
          style={
            [
              {
                "alignSelf": "stretch",
                "color": "#DEDDDE",
                "flex": 1,
                "fontFamily": "Helvetica Neue",
                "fontSize": 26,
                "height": 80,
                "marginHorizontal": 17,
                "paddingHorizontal": 0,
                "paddingVertical": 0,
                "textAlignVertical": "center",
              },
              false,
              undefined,
              undefined,
              undefined,
            ]
          }
          textAlignVertical="top"
          underlineColorAndroid="rgba(0, 0, 0, 0)"
          value=""
        />
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessibilityHint="save location"
        accessibilityLabel="save button"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            [
              {
                "alignItems": "center",
                "borderRadius": 6,
                "flexDirection": "row",
                "justifyContent": "center",
                "minHeight": 80,
                "overflow": "hidden",
                "paddingHorizontal": 17,
                "paddingVertical": 17,
              },
              {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#B6ACA6",
                "borderWidth": 1,
              },
            ],
            [
              {
                "backgroundColor": "#0B3D91",
                "borderRadius": 40,
                "borderWidth": 0,
                "height": 80,
                "marginVertical": 35,
                "width": "100%",
              },
              {
                "opacity": 0.5,
              },
            ],
            false,
          ]
        }
        testID="saveButton"
      >
        <Text
          allowFontScaling={false}
          style={
            [
              {},
              [
                {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                {
                  "fontFamily": "Helvetica Neue",
                },
                {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              [
                {
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "fontFamily": "HelveticaNeue-Medium",
                  "fontSize": 23,
                  "lineHeight": 29,
                  "textAlign": "center",
                  "zIndex": 2,
                },
                {
                  "color": "#FFFFFF",
                  "fontFamily": "HelveticaNeue-Medium",
                  "fontSize": 26,
                  "lineHeight": 30,
                },
                false,
              ],
            ]
          }
        >
          settings.locationSettingsData.addNewLocation.saveButton undefined
        </Text>
      </View>
    </View>
  </View>
</div>
`;
